unit2-java/lesson04.rst

```diff
@@ -0,0 +1,25 @@
+Unit 2 (Java) - Lesson 4 Learning Objectives
+============================================
+
+Goals
+-----
+
+- Use modifiers to customize fields and methods
+- Override built-in methods such as ``toString`` and ``equals``
+- Understand equality and sameness in Java
+
+Objectives
+----------
+
+- Explain and properly use the ``static`` keyword for fields and methods
+- Explain and properly use the ``final`` keyword
+- Create constant fields using ``static final``
+- Explain why instance variables may not be referenced within static methods
+- Write a custom ``toString`` method for a given class
+- Explain the concepts of object identity and equality, and how they differ
+- Explain and properly use ``.equals`` to determine when two objects are equal
+- Explain the behavior of ``==`` with respect to objects in Java
+- Describe and implement the best practices for implementing ``.equals`` within a class
+- Explain when ``.hashCode`` should be implemented within a class
+- Use IntelliJ’s code generation tools to generate well-structured ``.equals`` and ``.hashCode`` methods
+- Use the Single Responsibility Principle to create small, well-designed classes
```

Comment:
looks great! my only question would be is this more fitting for lesson 3? or are there concepts introduced here in 4 that warrant pushing this into this lesson?

Reply:
Good call. I'll move SRP up into lesson 3.